### ISSUE TITLE
Bug: Metadata and Options Parameter "NameIDFormat" not used correctly

### ISSUE THREAD

**Issue by MGraefe** [tngan/samlify#262]
When I specify a NameIDFormat inside my metadata like so: 

    <md:SPSSODescriptor>
      ...
      <md:EntityDescriptor>
        ...
        <md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified</md:NameIDFormat>
        ...

Then samlify still specfies `...nameid-format:emailAddress` in the request, regardless if it's a post or redirect binding.

According to the documentation, the ServiceProvider has an option named 'nameIDFormat' which does not alter the faulty behaviour if set to 'unspecified'. 

The only thing that currently fixes the problem is specifying `loginNameIDFormat: 'unspecified'` as an option in the constructor of the ServiceProvider.

'loginNameIDFormat' is only specified two times throughout the whole source code:

    $ grep -nr "loginNameIDFormat" *
    binding-post.ts:43:        NameIDFormat: namespace.format[spSetting.loginNameIDFormat] || namespace.format.emailAddress,
    binding-redirect.ts:95:        NameIDFormat: namespace.format[spSetting.loginNameIDFormat] || namespace.format.emailAddress,

so I think it's a typo, although I'm not sure that replacing loginNameIDFormat with nameIDFormat would actually fix the first issue (NameIDFormat not being read from the metadata file): While debugging I don't see loginNameID set as a member of the spSetting object even if it's specified in the metadata file.

__TLDR:__
- There likely is a typo in binding-post.ts:43 and binding-redirect.ts:95 it says `loginNameIDFormat` instead of `nameIDFormat`
  - looks like an easy fix
- NameIDFormat isn't recognized in the metadata file, instead if the constructor option of ServiceProvider isn't set it defaults to 'emailAddress' (don't know enough about the structure of the library to make the modification myself).
  - doesn't look as easy to fix, I don't know enough about the internal structure of the library to do this

**Comment by tngan** [comment]
@MGraefe See https://github.com/tngan/samlify/issues/284

**Comment by tngan** [comment]
v2.5.1 is released to npm.
